interview-questions.md

```diff
@@ -8,42 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: Implicit return means that even if you do not state a return Ruby will read the last line of code in and return that.
+
+Researched answer: In JavaScript a function needs to have a return statement in order for it to have a value for the return. In Ruby there is no need to use the "return" in order to recive an output when a method is ran this is what is known as implicit return and this occurs due to Ruby evaluating the last statement before "end" and returning that. If the progammer where to use "return" that would be known as an explicit return but the result would be exactly the same of implicit and explicit return for the method.
 
-Researched answer:
+https://franzejr.github.io/best-ruby/idiomatic_ruby/implicit_return.html
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: A block is a function that is passed to a method to determine an outcome.
 
-Researched answer:
+Researched answer: A block is a block of code that is used to get a return that is diffrent from the function but typically utilizes some of the data that is processed in the orginal function. Blocks are invoked by using the same name of the function. The logic or code of the block is always enclosed in curly braces ex. block ({}). You can also pass parameters into a block by using the word "yield" in the function that corresponds to the block if you want to pass more than one parameter you would use ||'s.
+
+https://www.tutorialspoint.com/ruby/ruby_blocks.htm#:~:text=Similarly%2C%20Ruby%20has%20a%20concept,as%20that%20of%20the%20block.
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer:To extract a value from a Ruby hash you will need to state the name of the hash and followed by square brackets and the key name inside of the brackets this will give access to the value that belongs to that key.
+
+Researched answer: It is a data type in Ruby that allows to store data by key-pair values that cna be easily acessed by the key. The structure of a hash is cat = {"name" => "Maki", "age" => 1} as you can see the key and value is separated by the arrow syntax. To call the value from the key you would use the following notation puts cat ["name"] that should output "Maki". Hashes such like arrays have built-on methods that can be used to manipulate the data that is stored whitin the hash.
 
```

Comment:
Try to elaborate more on your answers! You can go more in depth, give examples, and/or give analogies. The main goal is to fully convey the extent of your knowledge to the interviewer!